cli/cmd/lql_library.go

```diff
@@ -0,0 +1,109 @@
+//
+// Author:: Salim Afiune Maya (<afiune@lacework.net>)
+// Copyright:: Copyright 2020, Lacework Inc.
+// License:: Apache License, Version 2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package cmd
+
+import (
+	"github.com/lacework/go-sdk/api"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+var (
+	queryListLibraryCmd = &cobra.Command{
+		Use:   "list-library",
+		Short: "List queries from library",
+		Long:  `List all LQL queries in your Lacework Content Library.`,
+		Args:  cobra.NoArgs,
+		RunE:  listQueryLibrary,
+	}
+	queryShowLibraryCmd = &cobra.Command{
+		Use:   "show-library <query_id>",
+		Short: "Show a query from library",
+		Long:  `Show a query in your Lacework Content Library.`,
+		Args:  cobra.ExactArgs(1),
+		RunE:  showQueryLibrary,
+	}
+)
+
+func init() {
+	queryCmd.AddCommand(queryListLibraryCmd)
+	queryCmd.AddCommand(queryShowLibraryCmd)
+}
+
+func getListQueryLibraryTable(queries map[string]LCLQuery) (out [][]string) {
+	for id := range queries {
+		out = append(out, []string{id})
+	}
+	return
+}
+
+func listQueryLibrary(_ *cobra.Command, args []string) error {
+	cli.Log.Debugw("listing queries from library")
+
+	cli.StartProgress(" Retrieving queries...")
+	lcl, err := LoadLCL(*cli.LwComponents)
```

Comment:
why this is not like
```suggestion
	lcl, err := cli.LwComponent.LoadLCL()
```

Reply:
`LwComponents` is a pointer to `lwcomponent.State`:
```
LwComponents *lwcomponent.State
```

We don't want `LoadLCL` to be a method of `lwcomponent.State`, but we could make this a method of `cliState`...

Thoughts?

Reply:
Going to merge this down, but let's continue this discussion...